plugins/router/f5/f5.go

```diff
@@ -0,0 +1,1743 @@
+		Code string `json:"apiAnonymous"`
+	}{
+		Name: iRuleName,
+		Code: iRule,
+	}
+
+	err = f5.post(iRulesUrl, iRulePayload, nil)
+	if err != nil {
+		return err
+	}
+
+	glog.V(4).Infof("IRule %s created.", iRuleName)
+
+	return nil
+}
+
+func (f5 *f5LTM) ensureVserverHasIRule(vserverName, iRuleName string) error {
+	glog.V(4).Infof("Checking whether vserver %s has iRule %s...",
+		vserverName, iRuleName)
+
+	vserverUrl := fmt.Sprintf("https://%s/mgmt/tm/ltm/virtual/%s",
+		f5.host, vserverName)
+
+	res := struct {
+		Rules []string `json:"rules"`
+	}{}
+
+	err := f5.get(vserverUrl, &res)
+	if err != nil {
+		return err
+	}
+
+	commonIRuleName := fmt.Sprintf("/Common/%s", iRuleName)
+
+	for _, name := range res.Rules {
+		if name == commonIRuleName {
+			glog.V(4).Infof("Vserver %s has iRule %s associated with it;"+
+				" nothing to do.",
+				vserverName, iRuleName)
+			return nil
+		}
+	}
+
+	glog.V(4).Infof("Adding iRule %s to vserver %s...", iRuleName, vserverName)
+
+	vserverRulesPayload := struct {
+		Rules []string `json:"rules"`
+	}{
+		Rules: []string{sslPassthroughIRuleName},
+	}
+
+	err = f5.patch(vserverUrl, vserverRulesPayload, nil)
+	if err != nil {
+		return err
+	}
+
+	glog.V(4).Infof("IRule %s added to vserver %s.", iRuleName, vserverName)
+
+	return nil
+}
+
+// Initialize ensures that OpenShift-specific configuration is in place
+// on the F5 BIG-IP host.  In particular, this method creates a traffic
+// policy and associates it with the vservers if necessary.
+func (f5 *f5LTM) Initialize() error {
+	err := f5.ensurePolicyExists(httpPolicyName)
+	if err != nil {
+		return err
+	}
+
+	if f5.httpVserver != "" {
+		err = f5.ensureVserverHasPolicy(f5.httpVserver, httpPolicyName)
+		if err != nil {
+			return err
+		}
+	}
+
+	if f5.httpsVserver != "" {
+		err = f5.ensurePolicyExists(httpsPolicyName)
+		if err != nil {
+			return err
+		}
+
+		err = f5.ensureVserverHasPolicy(f5.httpsVserver, httpsPolicyName)
+		if err != nil {
+			return err
+		}
+
+		if f5.httpVserver != "" {
+			err = f5.ensureVserverHasPolicy(f5.httpVserver, httpPolicyName)
+			if err != nil {
+				return err
+			}
+		}
+
+		err = f5.ensureDatagroupExists(passthroughRoutesDataGroupName)
+		if err != nil {
+			return err
+		}
+
+		err = f5.ensureDatagroupExists(passthroughHostsDataGroupName)
+		if err != nil {
+			return err
+		}
+
+		err = f5.ensureIRuleExists(sslPassthroughIRuleName, sslPassthroughIRule)
+		if err != nil {
+			return err
+		}
+
+		err = f5.ensureVserverHasIRule(f5.httpsVserver, sslPassthroughIRuleName)
+		if err != nil {
+			return err
+		}
+	}
+
+	glog.V(4).Infof("F5 initialization is complete.")
+
+	return nil
+}
+
+// Create a pool named poolname on F5.
+func (f5 *f5LTM) CreatePool(poolname string) error {
+	url := fmt.Sprintf("https://%s/mgmt/tm/ltm/pool", f5.host)
+
+	payload := struct {
+		Mode    string `json:"loadBalancingMode"`
+		Monitor string `json:"monitor"`
+		Name    string `json:"name"`
+	}{
+		Mode:    "round-robin",
```

Comment:
One of the things that has been discussed is refactoring the router's internal model to use api objects.  At that point we'd be able to send hints to plugins using annotations that could drive items like this without doing a lot of extensions specific to the underlying implementations.  For instance we could have `openshift.io/f5/balance=round-robin` and `openshift.io/haproxy/balance=leastconn`.